bases/renku_data_services/data_api/main.py

```diff
@@ -26,6 +27,30 @@
 from renku_data_services.utils.middleware import validate_null_byte
 
 
+async def _send_messages() -> None:
+    config = Config.from_env()
+    while True:
+        try:
+            await config.event_repo.send_pending_events()
+            await asyncio.sleep(1)
+        except (asyncio.CancelledError, KeyboardInterrupt) as e:
+            logger.warning(f"Exiting: {e}")
+            return
+        except Exception as e:
+            logger.warning(f"Background task failed: {e}")
+            raise
+
+
+def send_pending_events() -> None:
+    """Send pending messages in case sending in a handler failed."""
+    app = Sanic("send_events")  # we need a dummy app for logging to work.  # noqa: F841
```

Comment:
Would this work?

```suggestion
   Sanic("send_events")  # we need a dummy app for logging to work.
```

Reply:
i can give it a try, just copied that from your earlier PR :slightly_smiling_face:

Reply:
😄 I just pushed the proof of concept for creating a managed background process and how to establish a communication channel. It wasn't production-ready code yet.

My intuition on this is that the logger may want access to `current_app` and that's why the line is needed here, btw.